cns/restserver/ipam.go

```diff
@@ -76,6 +97,54 @@ func (service *HTTPRestService) requestIPConfigHandler(w http.ResponseWriter, r
 	logger.ResponseEx(service.Name+operationName, ipconfigRequest, reserveResp, reserveResp.Response.ReturnCode, err)
 }
 
+var errNilStateStore = errors.New("nil endpoint state store")
+
+func (service *HTTPRestService) updateEndpointState(ipconfigRequest cns.IPConfigRequest, podInfo cns.PodInfo, podIPInfo cns.PodIpInfo) error {
+	if service.EndpointStateStore == nil {
+		return errNilStateStore
+	}
+	service.Lock()
+	defer service.Unlock()
+	logger.Printf("[updateEndpointState] Updating endpoint state for infra container %s", ipconfigRequest.InfraContainerID)
+	if endpointInfo, ok := service.EndpointState[ipconfigRequest.InfraContainerID]; ok {
+		logger.Printf("[updateEndpointState] Found existing endpoint state for infra container %s", ipconfigRequest.InfraContainerID)
```

Comment:
when do we get this case? if code hits here, isnt something wrong?

Reply:
I don't have a case in mind, but it is generally better to check for duplicates in a map?